### ISSUE TITLE
Issue with Chinese and Japanese characters

### ISSUE THREAD

**Issue by arthur-feral** [beaugunderson/emoji-aware#5]
Hello,

i'm using your package to parse emojis for a web application.
I encountred an issue in production for customers using chinese and japanese language.

here is an example with chinese
```javascript
var ea = require('emoji-aware');
var str = '你好，找文章';
var str2 = '你好找文章';

ea.split(str);
// false

ea.split(str);
// [ '你', '好', '找', '文', '章' ]
```

it looks like the middle char `，` is messing the process

same probleme with japanses:
```javascript
var str = '毎朝６時半に起きます';
var str2 = '毎朝時半に起きます';

ea.split(str);
// false

ea.split(str);
// [ '毎', '朝', '時', '半', 'に', '起', 'き', 'ま', 'す' ]
```
it works without the `6` char

here is another example
```javascript
var str = '、パリ郊外のアニエールにあるルイ･';
var str2 = '、パリ郊外のアニエールにあるルイ';

ea.split(str);
// false

ea.split(str);
// [ '、',  'パ',  'リ',  '郊',  '外',  'の',  'ア',  'ニ',  'エ',  'ー',  'ル',  'に',  'あ',  'る',  'ル', 'イ' ]
```

Maybe it is the order in a parser you created (https://github.com/beaugunderson/emoji-aware/blob/master/parsers/emoji.js#L53) ?

The other problem, it that your method `split` returns an array if everything is ok, and false otherwise.
It is a bad pattern because we expect the return to be an array, and in some case we got a boolean.
if `result.status === false`, it should return at least an empty array, or an array containing all chars (`string.split('')`)

So can you help me  understanding why these languages are barely supported ?

Thanks a lot

**Comment by beaugunderson** [comment]
sorry for the slow reply--I'll take a look at these issues soon!

**Comment by beaugunderson** [comment]
this is my fault; i was not including the characters after the surrogate pairs (those that end at `DF`):

![image](https://user-images.githubusercontent.com/61791/30926744-04773700-a36b-11e7-9058-73f0234e0931.png)

fixed in 3.0.3 ✨
